feat(triage signals): Remove fixability from run_automation and put into _generate_summary

Mihir-Mavalankar opened the pull request:
## PR Details
+ Removing fixability from run_automation to enable calling run_automation directly as discussed in meeting with @kddubey @roaga .
+ Updated the diagram too: https://miro.com/app/board/uXjVJqn1-fQ=/?focusWidget=3458764648325594380
+ Currently automation flow is unchanged. Fixability is still called before run_automation and after issue summary.

Mihir-Mavalankar committed:
first commit

Mihir-Mavalankar commented on src/sentry/seer/autofix/issue_summary.py:
```diff
@@ -303,18 +303,13 @@ def _run_automation(
+    fixability_score = group.seer_fixability_score
+    if not fixability_score:
+        logger.error("Fixability score is not available for group %s", group.id)
+        return
```
Fixed.

JoshFerge reviewed (dismissed):
please follow guidelines in our [guide](https://develop.sentry.dev/engineering-practices/commit-messages/) for naming pull requests please.

Mihir-Mavalankar committed:
if condition change

Mihir-Mavalankar commented on src/sentry/seer/autofix/issue_summary.py:
```diff
@@ -390,6 +383,22 @@ def _generate_summary(
+            logger.exception(
+                "Error generating fixability score in summary", extra={"group_id": group.id}
+            )
+
```
Okay I tried "Fix in Cursor" and it says it's not a bug either. Guess cursor bugbot screws up too.

roaga commented on src/sentry/seer/autofix/issue_summary.py:
```diff
@@ -303,18 +303,13 @@ def _run_automation(
+    fixability_score = group.seer_fixability_score
+    if fixability_score is None:
+        logger.error("Fixability score is not available for group %s", group.id)
+        return
```
you're planning on splitting run_automation out from issue summary, right? In that case, should we re-generate the summary/fixability score if it's not present when we reach this point?

Mihir-Mavalankar replied:
To support the original flow run_automation still needs to be in _generate_issue_summary. 
We can re-try here but if it doesn't exist then the endpoint is down/unreachable and we should cancel that automation run.

roaga replied:
but you're planning on calling generate_issue_summary on event #1 and run_automation on event #10 right? so they will eventually be called separately, so we should have the retry mechanism probably

and to support the original flow, run_automation can just be called immediately after generate_summary completes, no?

Mihir-Mavalankar replied:
Okay added a re-try back in run_automation.

Mihir-Mavalankar committed:
more tests for coverage

Mihir-Mavalankar committed:
add fixability check with better error log

Mihir-Mavalankar committed:
updated test

Mihir-Mavalankar committed:
added fixability re-try in run_automation

Mihir-Mavalankar commented on src/sentry/seer/autofix/issue_summary.py:
```diff
@@ -303,18 +303,22 @@ def _run_automation(
 
-    group.update(seer_fixability_score=issue_summary.scores.fixability_score)
+        group.update(seer_fixability_score=fixability_response.scores.fixability_score)
+        fixability_score = fixability_response.scores.fixability_score
```
That check I will have in the job that triggers the direct call to run_automation.

roaga replied:
oh i mean in that case we could also keep the fixability score check there too instead of putting the retry in here

Mihir-Mavalankar replied:
So you mean remove it from run_automation and add it in the task at directly calls run_automation? I did think of that and there I think if issue summary doesn't exist I was just gonna call _generate_summary which would do all 3 generate_summary, generate fixability and start_autofix.

Mihir-Mavalankar replied:
No I don't think putting  fixability and issue summary inside run_automation is a good idea. That function should just be to trigger automation. 

Ideally, I don't even want this re-try in run_automation. Even in the new flow, run_automation will never be called without issue_summary running first so we will always have fixability. run_automation will fail only if the call to that fixability endpoint fails which is true even today.

roaga replied:
sure if you do the retry before calling run_automation, that works too

Mihir-Mavalankar committed:
Update src/sentry/seer/autofix/issue_summary.py

Co-authored-by: Rohan Agarwal <47861399+roaga@users.noreply.github.com>

Mihir-Mavalankar committed:
Revert "added fixability re-try in run_automation"

This reverts commit b18d38cabd36e1426dddd80b28af8535a5b9f673.

jennmueng reviewed (approved):
after reviewing the diagram and looking at the threads this makes sense to me, although not fully in context with the discussions this seems like it wouldn't break the current flow.

Mihir-Mavalankar committed:
fix(tests): Update test expectation for high fixability score

With fixability score of 0.80, the stopping point should be OPEN_PR (not CODE_CHANGES)
because 0.80 >= 0.78 (SUPER_HIGH threshold of 0.76 + 0.02 buffer).
